src/chainparams.cpp

```diff
@@ -145,8 +145,9 @@ class CElementsParams : public CChainParams {
         parentGenesisBlockHash = uint256S("000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943");
 
         // Generate pegged Bitcoin asset
+        std::vector<unsigned char> commit = CommitToArguments(consensus, strNetworkID, genesisChallengeScript);
         uint256 entropy;
-        GenerateAssetEntropy(entropy, COutPoint() /* blank prevout to simulate coinbase */, parentGenesisBlockHash);
+        GenerateAssetEntropy(entropy, COutPoint(uint256(commit), 0) /* asset type commits to network params */, parentGenesisBlockHash);
```

Comment:
I still find the comments between function arguments very confusing visually. just bike-shedding.